### PR TITLE
Unit created

### PR DESCRIPTION
**Pull request by phlax**
currently the info stored about unit creation in submissions is patchy and basically redundant

this pr will

- stop creating UNIT_CREATE submissions
- add tests to ensure that proper creation information is recorded
- add a migration to remove all UNIT_CREATE submissions

### CODE REVIEW

File: pootle/apps/pootle_store/utils.py [hunk 1]
```diff
@@ -316,16 +316,6 @@ def sub_unmute_qc(self, **kwargs):
         _kwargs.update(kwargs)
         return self.create_submission(**_kwargs)
 
-    def sub_create(self, **kwargs):
```
Comment (unho) [inline pootle/apps/pootle_store/utils.py]: I find it odd that this was only used in tests.

File: tests/pootle_store/utils.py [hunk 2]
```diff
@@ -49,17 +50,9 @@ def test_unit_lifecycle_create(store0):
...
-    assert sub_create.translation_project == store0.translation_project
-    assert sub_create.revision == unit.revision
-    assert sub_create.type == SubmissionTypes.UNIT_CREATE
-    assert sub_create.field == SubmissionFields.TARGET
-    assert sub_create.new_value == unit.target
-    assert sub_create.old_value == ""
-    assert not sub_create.pk
+    source = unit.unit_source.get()
+    assert source.created_by == get_system_user()
+    assert source.created_with == SubmissionTypes.SYSTEM
```
Comment (unho) [inline tests/pootle_store/utils.py]: I am not sure if it appropriate to abuse the SubmissionTypes to specify a creation type.

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+from django.db import migrations
+
+
+logger = logging.getLogger(__name__)
+
+
+def remove_create_subs(apps, schema_editor):
+    subs = apps.get_model("pootle_statistics.Submission").objects.all()
+    to_delete = subs.filter(type=10)
```
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: I would a comment telling that this is the value for `SubmissionTypes.UNIT_CREATE`.

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    total = to_delete.count()
+    offset = 0
+    step = 10000
+    start = time.time()
+
+    while True:
+        units = to_delete[:step]
```
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: `units`? This is submissions.

File: pootle/apps/pootle_store/utils.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/utils.py]: the unitlifecycle class is not yet used but will be when i finish cleaning up subs

File: tests/pootle_store/utils.py (hunk 2 above)
Comment (phlax) [inline tests/pootle_store/utils.py]: that is what we currently use to record this information

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py (hunk 4 above)
Comment (phlax) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: yep

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        to_delete.filter(id__in=set(units.values_list("id", flat=True))).delete()
```
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: Is it not possible to just `units.delete()` ?

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+        logger.debug(
+            "deleted %s/%s in %s seconds"
+            % (offset + step, total, (time.time() - start)))
+        if offset > total:
+            break
+        offset = offset + step
```
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: Shouldn't the counter be incremented first and then checked?

**Comment by unho** [comment]
Is `SubmissionTypes.UNIT_CREATE` still needed?

File: tests/pootle_store/utils.py (hunk 2 above)
Comment (unho) [inline tests/pootle_store/utils.py]: Yes, on Submission, not on UnitSource.

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py (hunk 4 above)
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: Something like `submissions_chunk` sounds more explanatory.

**Comment by phlax** [comment]
> Is SubmissionTypes.UNIT_CREATE still needed?

no, i was just trying to remove it

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py (hunk 6 above)
Comment (phlax) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: no, then we would risk missing the last batch i think

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py (hunk 5 above)
Comment (phlax) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: no you cant delete a slice

File: pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py (hunk 6 above)
Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: I mean:

- we have 7 items to delete, offset is 0
- on first iteration:

  - it removes the 7 submissions,
  - 0 > 7 is False so doesn't break
  - then offset is incremented

- second iteration:

  - removes nothing
  - 10000 > 7 is True so aborts

Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: This applies to https://github.com/translate/pootle/commit/534be268120697269523032c79ea42133efb0570#diff-81b87d01574051b669a0370a4e83016dR42 as well.

Comment (unho) [inline pootle/apps/pootle_statistics/migrations/0012_remove_create_subs.py]: Yes, lets err on the safe side.

**Comment by unho** [comment]
lgtm

**Comment by ta2-1** [comment]
lgtm

**Pull request merged**